fido2-net-lib/MetadataService.cs

```diff
@@ -316,58 +317,35 @@ public sealed class MDSMetadata : IMetadataService
-        private MDSMetadata(string accessToken, string cachedirPath, System.Net.Http.HttpClient httpClient = null)
+        private static HttpClient _httpClient;
+        // Sample bogus key from https://fidoalliance.org/metadata/
+        private static readonly string _invalidToken = "6d6b44d78b09fed0c5559e34c71db291d0d322d4d4de0000";
+        private MDSMetadata(string accessToken, string cachedirPath, HttpClient httpClient = null)
         {
             // We need either an access token or a cache directory, but prefer both
             if (null == accessToken && null == cachedirPath) return;
-            
+
             // If we have only an access token, we can get metadata from directly from MDS and only cache in memory
             // If we have only a cache directory, we can read cached data (as long as it is not expired)
             // If we have both, we can read from either and update cache as necessary
             _accessToken = accessToken;
             _cacheDir = cachedirPath;
-            _httpClient = httpClient ?? new System.Net.Http.HttpClient();
+            _httpClient = httpClient ?? new HttpClient();
             if (null != _accessToken && 0x30 != _accessToken.Length && null != _cacheDir && 3 > _cacheDir.Length) throw new Fido2VerificationException("Either MDSAccessToken or CacheDir is required to instantiate Metadata instance");
-            
-            // Sample bogus key from https://fidoalliance.org/metadata/
-            var invalidToken = "6d6b44d78b09fed0c5559e34c71db291d0d322d4d4de0000";
-            if (_accessToken == invalidToken) conformance = true;
 
             payload = new System.Collections.Generic.Dictionary<Guid, MetadataTOCPayloadEntry>();
 
-            if (false == conformance)
-            {
-                // If we have a cache directory, let's try that first
-                if (true == System.IO.Directory.Exists(_cacheDir))
-                {
-                    try
-                    {
-                        GetTOCPayload(true);
-                    }
-                    catch (Exception ex)
-                    {
-                        if (ex is Fido2VerificationException || ex is System.IO.FileNotFoundException) { }
-                        else throw;
-                        // Something wrong with cached data, revert to MDS
-                    }
-                }
-                // If the payload count is still zero and we have what looks like a good access token, load from MDS
-                if (0 == payload.Count && null != _accessToken && 0x30 == _accessToken.Length)
-                {
-                    GetTOCPayload(false);
-                }
-                // If the payload count is zero, we've failed to load metadata
-                if (0 == payload.Count) throw new Fido2VerificationException("Failed to load MDS metadata");
-            }
-            else AddConformanceTOC();
+            if (false == ConformanceTesting()) Task.Run(AddMDSTOC);
```

Comment:
@aseigler I'm not completely sure but I feel some hesitation to create run a Task without support for awaiting and cancellation. I have an idea for this. What about removing the constructor for this class and let the consumer call a and await a Initialize method (if IsInitialized is false).

So: Let's add a two methods: `public async Task Initialize();` and a `public bool IsInitialized {get;}`

What do you think=

Reply:
Makes sense, definitely cleaner than this.

Reply:
This seems to work.  Caller can wait or timeout as they see fit, controllers currently should block until metadata is loaded.  Conformance test runs must block until all data is processed or tests will simply fail.

Reply:
@aseigler I'm wondering from where do you have this inverted `if` style e.g.  `if (0 == payload.Count)`? I never saw that in a C# repo.

Reply:
It's an old habit to avoid accidentally assigning a value when you meant to compare. If that's not what the kids are doing these days I can try to reform.

Reply:
@abergs I thought that comes from another programming language. For myself I'm used to the other style. The most important is, that everyone working on the repo is aware and agreed on a certain style :)

Reply:
I don't think style has ever come up, we've been too busy just trying to get this thing working and available.

Reply:
As @aseigler says, we've been busy making it work. Also (while i'm accustomed to it) sometimes that inverted if-statements made the cert/crypto code more readable!

I added a ticket here: https://github.com/abergs/fido2-net-lib/issues/87